### ISSUE TITLE
Add depth (bounces of the light) as parameter

### ISSUE THREAD

**Issue by Ale32** [raytracingDevTeam/raytracing#59]
It's definitely useful to have the _depth_ variable as a parameter as the other render options like _width_, _height_ and _samples_.

**Comment by andrearastelli** [comment]
To use this parameter, after some thinking, we may go in two different directions:

## 1 - Recursive approach with fixed value.
This approach is the more weird one, because in our current implementation we may go and define another parameter in the `color` function, and this parameter is always used as a constant.

Like this:
```c++
Color ray_color(const Ray &r, Hitable *world, int depth, const int max_depth)
{
    // ... some code that do bouncing stuff
    if (depth < max_depth)
    {
        return value1 + value2 * ray_color(r, world, depth + 1, max_depth);
    }
}
``` 

Something like this.
Is not really pretty, though, because when we eventually add multithreading we may want to convert this function from recursive to iterative to gain full control over the process.

## 2 - Change the function from recursive to iterative
Basically, the function will be updated into an iterative one, this way the process will be a little less weird, and will be easier to convert the application into a multithreaded one.
